FirebaseMessaging/Tests/UnitTests/FIRMessagingAuthKeychainTest.m

```diff
@@ -215,6 +215,11 @@ - (void)testKeyChainNoCorruptionWithUniqueService {
 }
 
 - (void)testQueryCachedKeychainItems {
```

Comment:
Rather than `#if` out the test implementation, could you regroup the tests so they can be excluded with a single  `#if`:
```objc
// Skip keychain tests on Catalyst and macOS. Tests are skipped because they
// involve interactions with the keychain that require a provisioning profile.
// See go/firebase-macos-keychain-popups for more details.
#if !TARGET_OS_MACCATALYST && !TARGET_OS_OSX

- (void)testQueryCachedKeychainItems { ... }

- (void)testCachedKeychainOverwrite { ... }

- (void)testSetKeychainItemShouldDeleteOldEntry { ... } 

- (void)testQueryAndAddEntry { ... }


#endif  // !TARGET_OS_MACCATALYST && !TARGET_OS_OSX
```

This is for two reasons, one to reduce the number of `#ifs` we use and two so, in the event we are ever debugging test logs, the only tests that show up were actually run.

Reply:
Made the change in the latest commit - afcd2c7dbc6b980127b716ac159089ef06f48612